### PR TITLE
Cleaning up use-sqs-for-backfill flag

### PR DESCRIPTION
**Pull request by KAbakumovAtl**
Cleaning up the flag.

As part of the cleanup we had to extract `findOrStartSync` function from `Subscription` class to a separate module. The reason for that is a cyclic dependency it was creating which was breaking tests.

It was moved to `sync-utils.ts`, also it makes more sense there, because this function relates more to sync than to subscriptions data model.

### CODE REVIEW

File: src/frontend/post-github-configuration.ts [hunk 1]
```diff
@@ -52,12 +53,15 @@ export default async (req: Request, res: Response): Promise<void> => {
...
 		}
 
-		await Subscription.install({
+		const subscription = await Subscription.install({
 			clientKey: getHashedKey(req.body.clientKey),
 			installationId: req.body.installationId,
 			host: res.locals.jiraHost
 		});
 
+		await findOrStartSync(subscription);
```
Comment (KAbakumovAtl) [inline src/frontend/post-github-configuration.ts:62]: Moved the logic of starting the sync outside the install function. This is to remove the cyclic dependency. Subscription class should be merely for data manipulation, and sync should be started externally.

File: src/sync/sync-utils.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		logger.info("Starting Jira sync");
+		await queues.discovery.add({ installationId, jiraHost });
+		return;
+	}
+
+
+	// Otherwise, just add a job to the queue for this installation
+	// This will automatically pick back up from where it left off
+	// if something got stuck
+	await sqsQueues.backfill.sendMessage({installationId, jiraHost}, 0, logger);
```
Comment (KAbakumovAtl) [inline src/sync/sync-utils.ts:38]: The function itself was moved from Subscription module here. 

Only this part of the function was changed by removing the feature flag.

**Comment by KAbakumovAtl** [comment]
I've separated queues which are used for sqs unit test and backfill tests. It helped to pass the build. However it is still a mystery to me why it was failing (and passing locally!).

It feels like that some sqs tests didn't wait to done() to be called. I'll add some logs to these tests to be prepared to analyse it further

**Review by bgvozdev — APPROVED** [review]

**Comment by KAbakumovAtl** [comment]
Just tested in staging, all good

File: .env.test [hunk 3]
```diff
@@ -14,6 +14,8 @@ SQS_DISCOVERY_QUEUE_URL=http://127.0.0.1:4566/000000000000/test-discovery
 SQS_DISCOVERY_QUEUE_REGION=us-west-1
 SQS_DEPLOYMENT_QUEUE_URL=http://127.0.0.1:4566/000000000000/test-deployment
 SQS_DEPLOYMENT_QUEUE_REGION=us-west-1
+SQS_TEST_QUEUE_URL=http://127.0.0.1:4566/000000000000/test-sqs-client
+SQS_TEST_QUEUE_REGION=us-west-1
```
Comment (thombergs) [inline .env.test:18]: What is this test queue for?

File: src/sync/installation.ts [hunk 4]
```diff
@@ -404,30 +405,27 @@ async function doProcessInstallation(app, job, installationId: number, jiraHost:
...
-	nextTaskDelays: Array<number>,
+	nextTaskDelaysMs: Array<number>,
 	logger: LoggerWithTarget
 ) {
-	if (nextTaskDelays.length > 0) {
-		nextTaskDelays.sort().reverse();
-		if (nextTaskDelays.length > 1) {
+	if (nextTaskDelaysMs.length > 0) {
+		nextTaskDelaysMs.sort().reverse();
+		if (nextTaskDelaysMs.length > 1) {
```
Comment (thombergs) [inline src/sync/installation.ts:414]: 👏🏼

File: src/sync/sync-utils.ts (hunk 2 above)
Comment (thombergs) [reply src/sync/sync-utils.ts:38]: Why not call the file `start-sync` instead of `sync-utils`?

File: .env.test (hunk 3 above)
Comment (thombergs) [reply .env.test:18]: Ah got it. It's used in integration tests.

**Review by thombergs — APPROVED** [review]

**Pull request merged**